packages/mobx/src/core/atom.ts

```diff
@@ -68,7 +68,7 @@ export class Atom implements IAtom {
      * Invoke this method _after_ this method has changed to signal mobx that all its observers should invalidate.
      */
     public reportChanged() {
-        if (globalState.inBatch && this.batchId_ !== globalState.batchId) {
+        if (!globalState.inBatch && this.batchId_ !== globalState.batchId) {
```

Comment:
&& => ||

Reply:
heh lol I should try to read before copy pasting next time. That fixes the test. However, the autorun doesn't initially reschedule itself (see the comment in the test), I think this used to be the case in the past, or am I mistaken here?

Reply:
> I think this used to be the case in the past, or am I mistaken here?

I don't know if this was the case in the past, but it wasn't the case when I wrote this test, otherwise there wouldn't be that extra `o.x++` outside the autorun to nudge it and it would fail, because `o.x` would end up being 6. I think I was probably a bit surprised as well when writing it. However, if there is some problem here, it was there before introducing this state version business.